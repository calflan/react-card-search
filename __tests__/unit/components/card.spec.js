import React from 'react';
import Card from '../../../src/js/components/card';
describe('Contact Component', () => {
  let component;
  let card;

  beforeEach(() => {
    card = {
      MoonpigProductNo: 1234,
      ProductImage: {
        Link: {
          Href: "http://someimage.com"
        }
      },
      Reviews: {
        ReviewCount: "1234"
      }
    }

    component = shallow(<Card card={card} />);
  });

  test('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('should set expandDetails in state to true when card image is clicked', () => {
    component.setState({ expandDetails: false });
    const cardImage = component.find('a').first();
    cardImage.simulate('click');
    expect(component.state().expandDetails).toBe(true);
  });
});