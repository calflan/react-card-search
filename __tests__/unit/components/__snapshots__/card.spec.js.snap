// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Component should match snapshot 1`] = `
<Fragment>
  <a
    className="card-image"
    href="#"
    onClick={[Function]}
    role="link"
  >
    <img
      role="img"
      src="http://someimage.com"
    />
  </a>
  <div
    class="card-listings__card-reviews"
  >
    <StarRatings
      changeRating={null}
      gradientPathName=""
      ignoreInlineStyles={false}
      name="rating"
      numberOfStars={5}
      rating={0}
      starDimension="20px"
      starEmptyColor="rgb(203, 211, 227)"
      starHoverColor="rgb(230, 67, 47)"
      starRatedColor="pink"
      starSpacing="0px"
      svgIconPath="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
      svgIconViewBox="0 0 51 48"
      typeOfWidget="Star"
    />
    <span>
      (
      1234
      )
    </span>
  </div>
</Fragment>
`;
